pkg/epp/flowcontrol/types/request.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
+
+import (
+	"context"
+	"time"
+)
+
+// FlowControlRequest is the contract for an incoming request submitted to the Flow Controller. It represents the "raw"
+// user-provided data and context for a single unit of work.
+//
+// An object implementing this interface is the primary input to `FlowController.EnqueueAndWait()`. The controller then
+// wraps this object with its own internal structures (which implement `QueueItemAccessor`) to manage the request's
+// lifecycle without modifying the original.
+type FlowControlRequest interface {
+	// Context returns the request's context. The Flow Controller uses this for monitoring cancellation (e.g., if the
+	// client disconnects or a request-scoped timeout occurs), which can lead to the request being evicted from a queue.
+	Context() context.Context
+
+	// FlowID returns the unique identifier for the flow this request belongs to (e.g., model name, tenant ID). The
+	// Flow Controller uses this ID, in conjunction with the flow's registered priority, to look up the active
+	// `ports.ManagedQueue` from the Flow Registry's `ports.RegistryShard`.
+	FlowID() string
+
+	// ByteSize returns the request's size in bytes (e.g., prompt size). This is used by the Flow Controller and for
+	// managing byte-based capacity limits and for Flow Registry statistics.
+	ByteSize() uint64
+
+	// InitialEffectiveTTL returns the suggested Time-To-Live for this request.
+	// This value is treated as a hint; the Flow Controller may override it based on its own configuration or policies.
+	// A zero value indicates the request has no specific TTL preference, and a system-wide default should be applied.
+	InitialEffectiveTTL() time.Duration
+
+	// ID returns an optional, user-facing unique identifier for this specific request. It is intended for logging,
+	// tracing, and observability. The core flow control logic does not use this ID for dispatching decisions; it uses
+	// the internal, opaque `QueueItemHandle`.
+	ID() string
+}
+
+// QueueItemHandle is an opaque handle to an item that has been successfully added to a `framework.SafeQueue`. It acts
+// as a key, allowing the Flow Controller to perform targeted operations (like removal) on a specific item without
```

Comment:
"it acts as a key".
this sounds a bit strange. I'd expect a `FlowControlRequest` to have a `RequestID` (or just `ID`) and use that identifier as the key for the request.
I've seen you defined ID in the Request but wrote it's optional and not used internally.

it's pretty hard to comment on code that still not here, but overall using an interface as a key instead of the ID sounds against the initial intuition.

might be good to get back to this point after seeing the implementation PR.